dags/web_preaggregated.py

```diff
@@ -0,0 +1,220 @@
+from datetime import datetime, timedelta
+import os
+from typing import Optional
+from collections.abc import Callable
+
+import dagster
+from dagster import Field, Array, Definitions
+
+from clickhouse_driver import Client
+from dags.common import ClickhouseClusterResource
+from posthog.models.web_preaggregated.sql import (
+    WEB_OVERVIEW_METRICS_DAILY_SQL,
+    DISTRIBUTED_WEB_OVERVIEW_METRICS_DAILY_SQL,
+    WEB_STATS_DAILY_SQL,
+    DISTRIBUTED_WEB_STATS_DAILY_SQL,
+    WEB_OVERVIEW_INSERT_SQL,
+    WEB_STATS_INSERT_SQL,
+)
+from posthog.clickhouse.cluster import ClickhouseCluster
+
+WEB_ANALYTICS_DAILY_PARTITION_DEFINITION = dagster.WeeklyPartitionsDefinition(
+    start_date="2025-01-01",
+    fmt="%Y-%m-%d",
+    timezone="UTC",
+)
+
+DEFAULT_TEAM_IDS = [int(id) for id in os.getenv("WEB_ANALYTICS_TEAM_IDS", "").split(",") if id]
+
+WEB_ANALYTICS_CONFIG_SCHEMA = {
+    "team_ids": Field(
+        Array(int),
+        default_value=DEFAULT_TEAM_IDS,
+        description="List of team IDs to process - leave empty to process all teams :fire:",
```

Comment:
How bad is it to run for all teams? I'd imagine we'd want to eventually (or all teams over a given size)

Reply:
It doesn't complete even for a week's worth of data because the in-memory parts of the ORDER BY reach our limit per host. We're taking a look at how we can do that by offloading to external memory  with better settings: https://clickhouse.com/docs/sql-reference/statements/select/group-by#group-by-in-external-memory

But I was adding the config just to test this first. The idea is to run for everyone. I am coming up with another dimension partition to run for a cluster of team_ids, so Dagster will handle that as well.